webgoat-integration-tests/src/test/java/org/owasp/webgoat/IntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+package org.owasp.webgoat;
+
+import io.restassured.RestAssured;
+import io.restassured.config.RestAssuredConfig;
+import io.restassured.config.SSLConfig;
+import lombok.Getter;
+import org.hamcrest.CoreMatchers;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.owasp.webwolf.WebWolf;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class IntegrationTest {
+
+    private static String WEBGOAT_URL = "http://localhost:8080/WebGoat/";
+    private static String WEBWOLF_URL = "http://localhost:9090/";
+
+    //This also allows to test the application with HTTPS when outside testing option is used
+    protected static RestAssuredConfig restConfig = RestAssuredConfig.newConfig().sslConfig(new SSLConfig().relaxedHTTPSValidation());
+
+    @Getter
+    private String webGoatCookie;
+    @Getter
+    private String webWolfCookie;
+    @Getter
+    private String webgoatUser = UUID.randomUUID().toString();
+
+    private static boolean started = false;
+
+    @BeforeClass
+    public static void beforeAll() {
+        if (!started) {
+            started = true;
+            if (!areAlreadyRunning()) {
+                SpringApplicationBuilder wgs = new SpringApplicationBuilder(StartWebGoat.class)
+                        .properties(Map.of("spring.config.name", "application-webgoat"));
+                wgs.run();
+                SpringApplicationBuilder wws = new SpringApplicationBuilder(WebWolf.class)
+                        .properties(Map.of("spring.config.name", "application-webwolf"));
+                wws.run();
+            }
+        }
+    }
+
+    private static boolean areAlreadyRunning() {
+        return checkIfServerIsRunnningOn(9090) && checkIfServerIsRunnningOn(8080);
+    }
+
+    private static boolean checkIfServerIsRunnningOn(int port) {
+        try (var ignored = new ServerSocket(port)) {
+            return false;
+        } catch (IOException e) {
+            return true;
+        }
+    }
+
+    protected String url(String url) {
+        url = url.replaceFirst("/WebGoat/", "");
+        url = url.replaceFirst("/WebGoat", "");
+        url = url.startsWith("/") ? url.replaceFirst("/", "") : url;
+        return WEBGOAT_URL + url;
+    }
+
+    protected String webWolfUrl(String url) {
+        url = url.startsWith("/") ? url.replaceFirst("/", "") : url;
+        return WEBWOLF_URL + url;
+    }
+
+    @Before
+    public void login() {
+        webGoatCookie = RestAssured.given()
```

Comment:
This is ok for the unique non-existing users, but the original steps would be required for an existing override user.

Reply:
correct, added the original steps